src/pdfFiller/1040FieldMappings.ts

```diff
@@ -0,0 +1,111 @@
+import { Dependent, FilingStatus, IncomeW2, Information, PersonRole } from '../redux/data'
+
+interface FieldMapping {
+  [field: number]: (info: Information) => string | boolean | undefined
+}
+
+const depField = (s: Information, pdfFieldIdx: number): string | boolean | undefined => {
+  const deps: Dependent[] = s.taxPayer?.dependents ?? []
+
+  // Based on the PDF row we are on, select correct dependent
+  const depIdx = Math.floor((pdfFieldIdx - 32) / 5)
+
+  if (depIdx < deps.length) {
+    const dep = deps[depIdx]
+    // Based on the PDF column, select the correct field
+    const depFieldIdx = (pdfFieldIdx - 32) % 5
+    return [`${dep.firstName} ${dep.lastName}`, dep.ssid, dep.relationship, false, false][depFieldIdx]
+  }
+
+  return undefined
+}
+
+// 1040 allows 4 dependents listed without a supplemental schedule,
+// so create field mappings for 4x5 grid of fields
+const depFieldMappings: FieldMapping = Object.fromEntries(
+  Array.from(Array(20)).map((u, n: number) => [n + 32, (s: Information) => depField(s, n + 32)])
+)
+
+const wages = (s: Information): number => {
+  if ((s.w2s ?? []).length < 1) {
+    return 0
+  }
+  return s.w2s.map((w2) => w2.income).reduce((l, r) => l + r)
```

Comment:
The wage strings are getting concatenated here rather than the incomes being summed

Reply:
Yep, all types are gone after compilation, so nothing stopping user input strings from going into fields assumed to be numbers.